### PR TITLE
Dashboard+ux

### PR DESCRIPTION
**Pull request by MaikNeubert**
* Charts are selectable (nothing you can do with currently)
* New Dashboard
* Display by items|boxes for Box Data
* from - to filter working (fill day function must be rewritten in the future to fill days before and after the first and last created boxes, so sometimes if you change the from - to filter it seems nothing happens)
* Responsive enough to not break completely on mobile

### CODE REVIEW

File: statviz/src/views/Statistics/components/DemographicChart.tsx [hunk 1]
```diff
@@ -23,32 +26,55 @@ export interface IDemographicCube {
...
+  if (facts.length === 0) {
+    return (
+      <Card w={props.width}>
+        <CardHeader>
+          <Heading size="md">Created Boxes</Heading>
+        </CardHeader>
+        <CardBody>
+          <p>
+            No demographic data available for your base. Either you are a
+            sending base which is not registering people or the birth rate is
```
Comment (pylipp) [inline statviz/src/views/Statistics/components/DemographicChart.tsx:46]: typo (?): birth date

**Comment by MaikNeubert** [comment]
>     * http://localhost:5173/bases/2/demographic shows an info message because no beneficiaries registered for that base
> 
>     * http://localhost:5173/bases/1/product-rank shows product names with NaN
> 
>     * http://localhost:5173/bases/1/created-boxes shows "no data for selected time range" but there's no filter

I created a new task in the project to work on the view of independend visualization


>     * http://localhost:5173/bases/1: dashboard view but no charts shown; even if I update the date filter, or the filter parameter in the URL

Are you sure there is data that falls in the selected time range? For me it is working as expected but i noticed i missed the "No Data" message for TopProducts if there is no data to display.

**Comment by pylipp** [comment]
@MaikNeubert false alarm from my side. I had tested it with the docker-compose db service which does not have history data. Hence probably the missing graphs.

I connected to the prod db and could get graphs with e.g. 
- http://localhost:5173/bases/2?boi=boxesCount&from=2019-01-01&to=2023-10-30
- http://localhost:5173/bases/32/demographic

**Review by pylipp — APPROVED** [review]

File: statviz/src/utils/hooks/useCreatedBoxes.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  CreatedBoxesResult,
+} from "../../types/generated/graphql";
+import { useMemo } from "react";
+import { createdBoxesTable } from "../table";
+import { useSearchParams } from "react-router-dom";
+
+export default function useCreatedBoxes(data?: {
+  createdBoxes: CreatedBoxesData;
+}) {
+  const [searchParams] = useSearchParams();
```
Comment (HaGuesto) [inline statviz/src/utils/hooks/useCreatedBoxes.ts:13]: maybe move data querying in here. see https://github.com/boxwise/boxtribute/blob/master/react/src/hooks/useMoveBoxes.ts

File: statviz/src/utils/table.ts [hunk 3]
```diff
@@ -69,5 +78,83 @@ export function table<Row extends object>(f: Array<Row>) {
...
+          ...row,
+        };
+      });
+
+      return table(joinedTable);
+    },
+  };
+}
+
+export function createdBoxesTable(createdBoxes: CreatedBoxesResult[]) {
```
Comment (HaGuesto) [inline statviz/src/utils/table.ts:102]: atm only used in useCreatedBoxes hook, so I do not see a clear reason yet why to separate this out from the hook

File: statviz/src/views/Statistics/components/filter/TimeRangeSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod/src/zod.js";
+import { useForm } from "react-hook-form";
+import { addDays, isBefore, subDays, subMonths } from "date-fns";
+import { useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+
+export const FilterCreatedOnFormScheme = z.object({
+  from: z
+    .date({
```
Comment (HaGuesto) [inline statviz/src/views/Statistics/components/filter/TimeRangeSelect.tsx:12]: maybe you want to check how zod works for custom validation (refine function) https://github.com/boxwise/boxtribute/blob/94da29211b2d7d5dd0db63930fb08eb448d1de15/react/src/views/BoxEdit/components/BoxEdit.tsx#L61

**Review by HaGuesto — COMMENTED** [review]
general recommendations:
- hooks folder in root 
- style folder in root
- components that are used in multiple views move into components folder and not views
- in view and components folder try to only have one tsx file representing a container (e.g. where data is prepared) and put all subcomponents in a subfolder

**Pull request merged**